src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -159,7 +159,7 @@ private static string EmitResponseWritingCall(this Endpoint endpoint)
     public static string EmitFilteredRequestHandler(this Endpoint endpoint, int baseIndent = 0)
     {
         var argumentList = endpoint.Parameters.Length == 0 ? string.Empty : $", {endpoint.EmitArgumentList()}";
-        var invocationConstructor = endpoint.Parameters.Length == 0 ? "DefaultEndpointFilterInvocationContext" : "EndpointFilterInvocationContext";
+        var invocationConstructor = endpoint.Parameters.Length == 0 || endpoint.Parameters.Length > 8 ? "new DefaultEndpointFilterInvocationContext" : "EndpointFilterInvocationContext.Create";
```

Comment:
(nit), sometimes this isn't a "constructor" anymore.

```suggestion
        var invocationCreator = endpoint.Parameters.Length == 0 || endpoint.Parameters.Length > 8 ? "new DefaultEndpointFilterInvocationContext" : "EndpointFilterInvocationContext.Create";
```

Reply:
For `endpoint.Parameters.Length == 0`, can't we just use `EndpointFilterInvocationContext.Create(httpContext)`?

Reply:
Yep, I've forgotten this in my reaction to the PR.